### PR TITLE
Build CSI driver in production merge-build codebuild job

### PR DESCRIPTION
**Pull request by sparrc**
### Summary

This change builds the CSI driver in the production codebuild "merge-build" job. This is similar to https://github.com/aws/amazon-ecs-agent/pull/3933

In addition to building, we also create artifacts in a similar format to the ecs-agent artifacts, with a "latest", "vX.XX.X", and "git hash" versioned artifacts. Lastly we make sha256 checksum hashes to be published.

Also bump up the version number of the codebuild image and change to privileged mode, to allow docker to run inside the codebuild container.

### Testing

This job was tested using a temporary codebuild job running merge-build.yml within the dev codebuild stack: https://github.com/aws/amazon-ecs-agent/pull/3947/files

### Description for the changelog

NA

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
  

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-infrastructure/release-pipeline-stack.yml [hunk 1]
```diff
@@ -401,9 +401,9 @@ Resources:
       Description: CodeBuild project to build the ECS Agent Docker image tarball and RPM
       Environment:
         ComputeType: BUILD_GENERAL1_SMALL
-        Image: aws/codebuild/amazonlinux2-x86_64-standard:3.0
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:5.0
```
Comment (fierlion) [inline build-infrastructure/release-pipeline-stack.yml:404]: do we want/need to update any of our other image versions?  Can you explain this version bump?
(same question for 2.0 -> 3.0 below)

**Review by fierlion — APPROVED** [review]

File: build-infrastructure/release-pipeline-stack.yml (hunk 1 above)
Comment (sparrc) [reply build-infrastructure/release-pipeline-stack.yml:404]: we don't need to bump any of them, this is just for housekeeping more than anything, we did the same for the pr-build jobs in the previous PR: https://github.com/aws/amazon-ecs-agent/pull/3933

**Review by amogh09 — APPROVED** [review]

File: buildspecs/merge-build.yml [hunk 2]
```diff
@@ -128,5 +147,13 @@ artifacts:
     # rpm
     - $ECS_AGENT_RPM
 
+    # csi driver
```
Comment (prateekchaudhry) [inline buildspecs/merge-build.yml:150]: do these artifacts need to be added to `signing.yml` script to get the pipeline working end to end?

Comment (sparrc) [reply buildspecs/merge-build.yml:150]: yes they do, i am working on that now

**Pull request merged**